### DISCUSSION TITLE
Test Fails at EXPECT_GE(sleep_time, 0);

### DISCUSSION THREAD

**Discussion by xMHW** [ANLAB-KAIST/KENSv3#210] [Project Q&A 2. Transport Layer (TCP)] [answered]
When I'm trying to run test-kens-transfer-**unreliable** test case TestTransfer_Connect_Send_Symmetric in my Linux machine I'm getting an error as follows
```
KENSv3/app/kens/testtransfer.cpp:248: Failure
Expected: (sleep_time) >= (0), actual: -536002 vs 0

KENSv3/app/kens/testtransfer.cpp:111: Failure
Expected: (sleep_time) >= (0), actual: -538003 vs 0
```
which corresponds to
```
    struct timeval tv;
    ret = gettimeofday(&tv, 0);
    EXPECT_EQ(ret, 0);

    long sleep_time = start_time - (1000 * 1000 * tv.tv_sec) - tv.tv_usec;
    EXPECT_GE(sleep_time, 0);
```
This code block (EXPECT_GE(sleep_time, 0))
It didn't happen for **reliable test** for the same case(Also, in my macOS machine, it doesn't fail either for unreliable case).
So, I don't think this is the problem of my implementation.
Could you help me get rid of this problem?
Otherwise, can I just comment out those lines? (BTW, after commenting out those two lines, I pass the test cases)

++
Also, it fails when I run test-kens-close-**unreliable** too. TestEnv_Any.TestClose_Connect_CloseFirst, TestEnv_Any.TestClose_Connect_CloseLater, TestEnv_Any.TestClose_Connect_CloseSimultaneous
```
KENSv3/app/kens/testclose.cpp:140: Failure
Expected: (sleep_time) >= (0), actual: -526002 vs 0

KENSv3/app/kens/testclose.cpp:86: Failure
Expected: (sleep_time) >= (0), actual: -518003 vs 0
```
Same as above, comment out then test pass.
Only happens for unreliable case.

Also, below is the full list of test cases where this error happens
```
test-kens-transfer-unreliable
[  FAILED  ] 6 tests, listed below:
[  FAILED  ] TestEnv_Any.TestTransfer_Connect_Send_Symmetric
[  FAILED  ] TestEnv_Any.TestTransfer_Connect_Send_EOF
[  FAILED  ] TestEnv_Any.TestTransfer_Connect_Recv_Symmetric
[  FAILED  ] TestEnv_Any.TestTransfer_Connect_Recv_EOF
[  FAILED  ] TestEnv_Any.TestTransfer_Connect_Recv_SmallBuffer1
[  FAILED  ] TestEnv_Any.TestTransfer_Connect_Recv_SmallBuffer2
test-kens-close-unreliable
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] TestEnv_Any.TestClose_Connect_CloseFirst
[  FAILED  ] TestEnv_Any.TestClose_Connect_CloseLater
[  FAILED  ] TestEnv_Any.TestClose_Connect_CloseSimultaneous
```

**Comment by Pusnow** [comment, marked as answer]
No, please do not comment out the lines.

It seems your connection establishment is too slow (maybe due to drops?).
Please try to shorten the connection establishment time.

**Reply by xMHW** [reply]
Ok, I'll try that out.
